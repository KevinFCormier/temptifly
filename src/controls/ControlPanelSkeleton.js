'use strict'

import React from 'react'
import PropTypes from 'prop-types'
import { SkeletonText } from 'carbon-components-react'

class ControlPanelSkeleton extends React.Component {
  static propTypes = {
    control: PropTypes.object,
    controlId: PropTypes.string
  };

  constructor(props) {
    super(props)
  }

  render() {
    const { controlId, control } = this.props
    const { name } = control

    return (
      <React.Fragment>
        <div className="creation-view-controls-skeleton">
          <label
            className="creation-view-controls-textbox-title"
            htmlFor={controlId}
          >
            {name}
          </label>
          <SkeletonText id={controlId} />
        </div>
      </React.Fragment>
    )
  }
}

export default ControlPanelSkeleton
