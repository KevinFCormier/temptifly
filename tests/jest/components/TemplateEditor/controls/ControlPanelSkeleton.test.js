'use strict'

import React from 'react'
import ControlPanelSkeleton from '../../../../../src/controls/ControlPanelSkeleton'
import renderer from 'react-test-renderer'

import { control } from '../../TestingData'

describe('ControlPanelSkeleton component', () => {
  it('renders as expected', () => {
    const fn = jest.fn()
    const component = renderer.create(
      <ControlPanelSkeleton
        key={'key'}
        control={control}
        controlId={'controlId'}
        handleChange={fn}
        i18n={fn}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
