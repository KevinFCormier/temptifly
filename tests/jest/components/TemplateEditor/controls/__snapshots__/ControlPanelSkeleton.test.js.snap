// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlPanelSkeleton component renders as expected 1`] = `
<div
  className="creation-view-controls-skeleton"
>
  <label
    className="creation-view-controls-textbox-title"
    htmlFor="controlId"
  >
    creation.app.name
  </label>
  <div
    className="pf-c-skeleton"
    id="controlId"
  >
    <span
      className="pf-u-screen-reader"
    />
  </div>
</div>
`;
